File: frontend/lib/models/joseki_api.dart
import 'package:frontend/models/joseki.dart';

class JosekiVideoRequest {
  final List<Stone> stones;
  final String videoId;

  JosekiVideoRequest({
    required this.stones,
    required this.videoId,
  });

  Map<String, dynamic> toJson() => {
    'joseki': {
      'stones': stones.map((stone) => stone.toJson()).toList(),
    },
    'video': {
      'id': videoId,
    },
  };
}

class StonesRequest {
  final List<Stone> stones;

  StonesRequest({
    required this.stones,
  });

  Map<String, dynamic> toJson() => {
    'stones': stones.map((stone) => stone.toJson()).toList(),
  };
}

File: web/src/models/josekiApi.ts
import type { Stone } from "./joseki"

export class JosekiVideoRequest {
  stones: Array<Stone>
  videoId: string

  constructor(stones: Array<Stone>, videoId: string) {
    this.stones = stones
    this.videoId = videoId
  }

  toJson() {
    return {
      joseki: {
        stones: this.stones.map((stone) => stone.toJson()),
      },
      video: {
        id: this.videoId,
      },
    }
  }
}

export class StonesRequest {
  stones: Array<Stone>

  constructor(stones: Array<Stone>) {
    this.stones = stones
  }

  toJson() {
    return {
      stones: this.stones.map((stone) => stone.toJson()),
    }
  }
}
